android/capacitor/src/main/java/com/getcapacitor/WebViewLocalServer.java

```diff
@@ -213,6 +213,10 @@ private WebResourceResponse handleLocalRequest(WebResourceRequest request, PathH
                 String[] parts = rangeString.split("=");
```

Comment:
Am just driving by while hunting out my own problem... I wonder if a better way to do this is to make the copy only when it's about to be modified here.

I.e., instead of creating a copy every access, do it when the `tempResponseHeaders` variable is assigned at line 208:
```
Map<String, String> tempResponseHeaders = new HashMap(handler.getResponseHeaders());
```

Other than that though, this code definitely makes good sense to me.

Reply:
I see what you mean (a getter should not create new objects), but this will re-introduce the problem, as there are other invocations of this method as well - all passing the returned instance into the constructor of `WebResourceResponse` (which is where the value of `tempResponseHeaders` also ends up). Note that it's not the headers set on lines 225/226 that cause trouble here, but the `Content-Length` header, set somewhere else downstream from this code.

We could of course move the cloning to all call points of the getter, but that leaves a trap for the next person coming around and wanting to use this method, probably not knowing about this issue, so we should probably not do it without making the map read-only (through [Collections.unmodifiableMap(...)](https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#unmodifiableMap-java.util.Map-), ie. `this.responseHeaders = Collections.unmodifiableMap(new LinkedHashMap<String, String>(tempResponseHeaders))` on line 102). The downside to this approach, though, is that it triggers runtime exceptions - and I can't find the automated tests that will trigger these when developing, leaving it to the developer to thoroughly test all cases before submitting a patch...

Maybe it would be better to rename `WebViewLocalServer.PathHandler.getResponseHeaders()` to `WebViewLocalServer.PathHandler.buildDefaultResponseHeaders()` to clarify? I don't think I can see a case where the set of response headers need to be shared between responses, only cases where there is a base set of headers to be included in every response.

What do you think, @peitschie ?